File: sorting/bubble_sort/bubble_sort.js
function bubble_sort(arr) {
  console.log("before:", arr);
  for (var i=arr.length-1; i>0; i-- ) {
    for (var j=0; j<i;j++) {
      if (arr[j] > arr[j+1]) {
        swap(arr,j,j+1);
      }
    }
  }
  return arr;
}

function swap(arr, a, b) {
    const temp = arr[a];
    arr[a] = arr[b];
    arr[b] = temp;
}


array = [5,6,2,4,3,1,0,0]

console.log("sorted:", bubble_sort(array))

File: sorting/bubble_sort/bubble_sort.rb
def bubble_sort(arr)
	len = arr.length
	for i in 0...len-1
		for j in 0...len-i-1
			if arr[j] > arr[j+1]
				temp = arr[j]
				arr[j] = arr[j+1]
				arr[j+1] = temp
			end
		end
	end
	return arr
end